bench/BenchAll.hs

```diff
@@ -79,14 +87,57 @@ smallIntegerData = map fromIntegral intData
 largeIntegerData :: [Integer]
 largeIntegerData = map (* (10 ^ (100 :: Integer))) smallIntegerData
 
+{-# NOINLINE floatPosData #-}
+floatPosData :: [Float]
+floatPosData = map evenlyDistribute intData
+  where
+  evenlyDistribute :: Int -> Float
+  evenlyDistribute x = castWord32ToFloat $ increment * fromIntegral x
+  increment = castFloatToWord32 maxFinite `div` fromIntegral nRepl
 
-{-# NOINLINE floatData #-}
-floatData :: [Float]
-floatData = map (\x -> (3.14159 * fromIntegral x) ^ (3 :: Int)) intData
+{-# NOINLINE floatNegData #-}
+floatNegData :: [Float]
+floatNegData = map negate floatPosData
 
-{-# NOINLINE doubleData #-}
-doubleData :: [Double]
-doubleData = map (\x -> (3.14159 * fromIntegral x) ^ (3 :: Int)) intData
+{-# NOINLINE floatSpecials #-}
+floatSpecials :: [Float]
+floatSpecials = foldMap (const specials) [1..nRepl `div` length specials]
+  where
+  specials = [nan, infinity, negate infinity, 0 -0]
+
+{-# NOINLINE doublePosData #-}
+doublePosData :: [Double]
+doublePosData = map evenlyDistribute intData
+  where
+  evenlyDistribute :: Int -> Double
+  evenlyDistribute x = castWord64ToDouble $ increment * fromIntegral x
+  increment = (maximum - minimum) `div` fromIntegral nRepl
+  minimum = castDoubleToWord64 $ succIEEE $ 2 ^ 53
+  maximum = castDoubleToWord64 maxFinite 
+
+{-# NOINLINE doubleNegData #-}
+doubleNegData :: [Double]
+doubleNegData = map negate doublePosData
+
+-- f is an integer in the range [1, 2^53).
```

Comment:
Wait, but this doesn't actually specifically generate integers, does it?

Reply:
I'm sorry, I do not know why I put that comment there any more but it does not generate integers.

Reply:
Humm. There is a special case for small integer `Double`s in the current code: see `d2dSmallInt` in `Data.ByteString.Builder.RealFloat.Internal`. So it may make sense to have a benchmark that exercises these and checks its worthwhile-ness.

Perhaps a decent way to get a variety of integers in that range is via something like `[fromInteger $ round $ sinh (c * fromIntegral k) / c | k <- [1..10000]]` for some appropriately chosen constant `c`?